MapboxNavigation/EndOfRouteViewController.swift

```diff
@@ -0,0 +1,193 @@
+import UIKit
+import MapboxDirections
+
+fileprivate enum ConstraintSpacing: CGFloat {
+    case closer = 8.0
+    case further = 65.0
+}
+
+fileprivate enum ContainerHeight: CGFloat {
+    case normal = 200
+    case commentShowing = 260
+}
+
+/// :nodoc:
+@objc(MBEndOfRouteContentView)
+open class EndOfRouteContentView: UIView {}
+
+/// :nodoc:
+@objc(MBEndOfRouteTitleLabel)
+open class EndOfRouteTitleLabel: StylableLabel {}
+
+/// :nodoc:
+@objc(MBEndOfRouteStaticLabel)
+open class EndOfRouteStaticLabel: StylableLabel {}
+
+/// :nodoc:
+@objc(MBEndOfRouteCommentView)
+open class EndOfRouteCommentView: StylableTextView {}
+
+/// :nodoc:
+@objc(MBEndOfRouteButton)
+open class EndOfRouteButton: StylableButton {}
+
+class EndOfRouteViewController: UIViewController {
+
+    //MARK: - IBOutlets
+    @IBOutlet weak var labelContainer: UIView!
+    @IBOutlet weak var primary: UILabel!
+    @IBOutlet weak var endNavigationButton: UIButton!
+    @IBOutlet weak var stars: RatingControl!
+    @IBOutlet weak var commentView: UITextView!
+    @IBOutlet weak var commentViewContainer: UIView!
+    @IBOutlet weak var showCommentView: NSLayoutConstraint!
+    @IBOutlet weak var hideCommentView: NSLayoutConstraint!
+    @IBOutlet weak var ratingCommentsSpacing: NSLayoutConstraint!
+    
+    //MARK: - Properties
+    lazy var placeholder: String = NSLocalizedString("How can we improve?", comment: "Comment Placeholder Text")
+    lazy var endNavigation: String = NSLocalizedString("End Navigation", comment: "End Navigation Button Text")
+    
+    var dismiss: ((Int, String?) -> Void)?
+    var comment: String?
+    var rating: Int = 0 {
+        didSet {
+            rating == 0 ? hideComments() : showComments()
+        }
+    }
+    
+    open var destination: Waypoint? {
+        didSet {
+            guard isViewLoaded else { return }
+            updateInterface()
+        }
+    }
+
+    //MARK: - Lifecycle Methods
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        clearInterface()
+        stars.didChangeRating = { (new) in self.rating = new }
```

Comment:
`RatingControl` have a strong ref to the `didChangeRating` closure so this is a strong ref cycle. This closure needs a weak self.

Reply:
- [ ] weakify self

Reply:
I'm not so sure this is a retain cycle, since `EndOfRouteViewController` only holds a weak reference to the `RatingControl`, but it was an oversight nonetheless.